infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSshKeysProvisioner.java

```diff
@@ -0,0 +1,218 @@
+                  .withName(getValidNameForSecret(sshPair.getName()))
+                  .endMetadata()
+                  .build();
+
+          k8sEnv.getSecrets().put(secret.getMetadata().getName(), secret);
+
+          k8sEnv
+              .getPodsData()
+              .values()
+              .forEach(p -> addSshKeySecret(secret.getMetadata().getName(), p.getSpec()));
+
+          sshConfig.append(buildConfig(sshPair.getName()));
+        }
+
+        map.put(SSH_CONFIG, sshConfig.toString());
+        ConfigMap configMap =
+            new ConfigMapBuilder()
+                .withNewMetadata()
+                .withName("sshconfigmap")
+                .endMetadata()
+                .withData(map)
+                .build();
+
+        k8sEnv.getConfigMaps().put(configMap.getMetadata().getName(), configMap);
+        k8sEnv.getPodsData().values().forEach(p -> addConfigFile(p.getSpec()));
+      }
+    } catch (ServerException e) {
+      LOG.warn("Unable get SSH Keys ", e);
+    }
+  }
+
+  private void addSshKeySecret(String secretName, PodSpec podSpec) {
+    podSpec
+        .getVolumes()
+        .add(
+            new VolumeBuilder()
+                .withName(secretName)
+                .withSecret(new SecretVolumeSourceBuilder().withSecretName(secretName).build())
+                .build());
+    List<Container> containers = podSpec.getContainers();
+    containers.forEach(
+        container -> {
+          VolumeMount volumeMount =
+              new VolumeMountBuilder()
+                  .withName(secretName)
+                  .withNewReadOnly(false)
+                  .withReadOnly(false)
+                  .withMountPath(SSH_BASE_CONFIG_PATH + secretName)
+                  .build();
+          container.getVolumeMounts().add(volumeMount);
+        });
+  }
+
+  private void addConfigFile(PodSpec podSpec) {
+    podSpec
+        .getVolumes()
+        .add(
+            new VolumeBuilder()
+                .withName("configvolume")
+                .withConfigMap(new ConfigMapVolumeSourceBuilder().withName("sshconfigmap").build())
+                .build());
+
+    List<Container> containers = podSpec.getContainers();
+    containers.forEach(
+        container -> {
+          VolumeMount volumeMount =
+              new VolumeMountBuilder()
+                  .withName("configvolume")
+                  .withMountPath(SSH_CONFIG_PATH)
+                  .withSubPath(SSH_CONFIG)
+                  .withReadOnly(false)
+                  .withNewReadOnly(false)
+                  .build();
+          container.getVolumeMounts().add(volumeMount);
+        });
+  }
+
+  /**
+   * @param host the host of version control service (e.g. github.com, gitlab.com and etc)
+   * @return the ssh configuration which include host of host and identity file location
+   *     <p>Example of provided configuration:
+   *     <p>host github.com HostName github.com IdentityFile /etc/ssh/github-com/ssh-privatekey
```

Comment:
Is it true? I see `\n` in code, it's only one line example.

Reply:
fixed